terraform/azure/modules/0_landing_zone/agw-public/variables.tf

```diff
@@ -0,0 +1,39 @@
+variable "resource_prefix" {
+  description = "Prefix used for naming all resources"
+  type        = string
+}
+
+variable "agw_resource_group_name" {
+  description = "The name of the resource group"
+  type        = string
+}
+
+variable "agw_vnet_name" {
+  description = "Name of vNet"
+  type        = string
+}
+
+variable "agw_subnet_name" {
+  description = "Subnet to deploy App Gateway in"
+  type        = string
+}
+
+variable "agw_key_vault_name" {
+  description = "Key Vault Name"
```

Comment:
Can this be a little more descriptive? Like does the key vault exist or is it being created. If it is being created should it use the resource_prefix plus standard naming?